### PR TITLE
Ensures that Gem Bundler is present before setup

### PR DESCRIPTION
**Pull request by luiseugenio**

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -1,8 +1,15 @@
 #!/usr/bin/env bash
-set -euo pipefail
-IFS=$'\n\t'
-set -vx
+BUNDLER_EXISTS=$(gem list -i bundler)
+if [ "$BUNDLER_EXISTS" = true ] ; then
```
Comment (conarro) [inline bin/setup:3]: Will this ensure the version is correct? It appears to just be checking for the gem's presence, but the related issue (#3 ) indicates a version requirement as well.

Comment (luiseugenio) [reply bin/setup:3]: Right. I updated the code now to check the bundler version as well.

Comment (conarro) [reply bin/setup:3]: Actually, just realized we should just remove `Gemfile.lock` from the repo. That shouldn't be tracked for gems, as we can't/shouldn't lock gem users into specific versions.

Want to update this PR to remove that file? And then revert this to do your basic `bundler` check without regard to version?

Comment (luiseugenio) [reply bin/setup:3]: If I remove `Gemfile.lock` and go back to the version without checking the `bundler` version (`bin/setup`), see what happens when we don't have the ~ 2.0 version installed.

```
Bundler could not find compatible versions for gem "bundler":
  In Gemfile:
    bundler (~> 2.0)

  Current Bundler version:
    bundler (1.17.3)
This Gemfile requires a different version of Bundler.
Perhaps you need to update Bundler by running `gem install bundler`?

Could not find gem 'bundler (~> 2.0)' in any of the relevant sources:
  the local ruby installation
```

Comment (conarro) [reply bin/setup:3]: Hm, well we need to remove `Gemfile.lock` either way. Can you update this PR with the removal and we can get it merged?

Comment (luiseugenio) [reply bin/setup:3]: Done!

**Pull request merged**